test/test_mcmc.py

```diff
@@ -138,19 +163,19 @@ def model(data):
     ])
     init_params, potential_fn, constrain_fn = initialize_model(random.PRNGKey(4), model, count_data)
     init_kernel, sample_kernel = hmc(potential_fn)
-    hmc_state = init_kernel(init_params, num_warmup=warmup_steps)
-    hmc_states = fori_collect(num_samples, sample_kernel, hmc_state,
-                              transform=lambda x: constrain_fn(x.z))
-    tau_posterior = (hmc_states['tau'] * len(count_data)).astype("int")
+    hmc_state = init_kernel(init_params, num_warmup=warmup_steps, dense_mass=dense_mass)
+    samples = fori_collect(num_samples, sample_kernel, hmc_state,
+                           transform=lambda x: constrain_fn(x.z))
+    tau_posterior = (samples['tau'] * len(count_data)).astype("int")
     tau_values, counts = onp.unique(tau_posterior, return_counts=True)
     mode_ind = np.argmax(counts)
     mode = tau_values[mode_ind]
     assert mode == 44
```

Comment:
I think that this is not a good test case, where posterior is multimodal and not continuous (energy will jump/drop at each bin `i / len(count_data)`). It might be better to run do multi-chain analysis on this problem (where we can remove bad chains or only select a small number of samples in bad chains based on their n_eff). For now, I think we can either choose another random seed, disable the test for some cases, or `assert mode in [12, 44,...]`

Reply:
I think this test is just too flaky, so I am exercising this code path on other tests now.

Reply:
Hmm..actually this is failing even after reverting my changes, so this seems to just be flaky. I'll just add an xfail marker and we can decide how best to handle this later.